addons/hr_timesheet/static/tests/hr_timesheet_common_tests.js

```diff
@@ -0,0 +1,136 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { session } from "@web/session";
+import { companyService } from "@web/webclient/company_service";
+
+import { patchWithCleanup } from "@web/../tests/helpers/utils";
+import { setupViewRegistries } from "@web/../tests/views/helpers";
+
+
+export const getServerData = () => JSON.parse(JSON.stringify({
+    models: {
+        'account.analytic.line': {
+            fields: {
+                project_id: { string: "Project", type: "many2one", relation: "project.project" },
+                task_id: { string: "Task", type: "many2one", relation: "project.task" },
+                unit_amount: { string: "Unit Amount", type: "integer" },
+            },
+            records: [
+                { id: 1, project_id: 1, task_id: 3, unit_amount: 1 },
+                { id: 2, project_id: 1, task_id: false, unit_amount: 1 },
+                { id: 3, project_id: false, task_id: false, unit_amount: 1 },
+            ],
+        },
+        'project.project': {
+            fields: {
+                name: { string: "Name", type: "string" },
+            },
+            records: [
+                { id: 1, name: "Project 1" },
+            ],
+        },
+        'project.task': {
+            fields: {
+                name: { string: "Name", type: "string" },
+                project_id: { string: "Project", type: "many2one", relation: "project.project" },
+            },
+            records: [
+                { id: 1, name: "Task 1\u00A0AdditionalInfo", project_id: 1 },
+                { id: 2, name: "Task 2\u00A0AdditionalInfo", project_id: 1 },
+                { id: 3, name: "Task 3\u00A0AdditionalInfo", project_id: 1 },
+            ],
+        }
+    },
+    views: {
+        "account.analytic.line,false,form": `
+            <form>
+                <field name="project_id"/>
+                <field name="task_id"/>
+                <field name="unit_amount"/>
+            </form>
+        `,
+        "account.analytic.line,false,list": `
+            <tree editable="bottom">
+                <field name="project_id"/>
+                <field name="task_id"/>
+                <field name="unit_amount"/>
+            </tree>
+        `,
+    },
+}));
+
+export function updateArch(serverData, fieldNameWidgetNameMapping = {}, fieldNameContextMapping = {}) {
+    for (const viewKey in serverData.views) {
+        for (const [fieldName, widgetName] of Object.entries(fieldNameWidgetNameMapping)) {
+            serverData.views[viewKey] = serverData.views[viewKey].replace(
+                `name="${fieldName}"`,
+                `name="${fieldName}" widget="${widgetName}"`
+            );
+        }
+        for (const [fieldName, context] of Object.entries(fieldNameContextMapping)) {
+            serverData.views[viewKey] = serverData.views[viewKey].replace(
+                `name="${fieldName}"`,
+                `name="${fieldName}" context="${context}"`
```

Comment:
If you want you could parse the JSON containing the context here instead of passing a JSON in a string as parameter. 🙂 
```suggestion
                `name="${fieldName}" context="${JSON.parse(context)}"`
```

Reply:
JSON.stringify*